components/admin_console/reset_password_modal/reset_password_modal.tsx

```diff
@@ -1,29 +1,50 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
+import {UserProfile} from 'mattermost-redux/types/users';
 
-import {adminResetPassword} from 'actions/admin_actions.jsx';
 import * as Utils from 'utils/utils.jsx';
 
-export default class ResetPasswordModal extends React.Component {
-    static propTypes = {
-        user: PropTypes.object,
-        currentUserId: PropTypes.string.isRequired,
-        show: PropTypes.bool.isRequired,
-        onModalSubmit: PropTypes.func,
-        onModalDismissed: PropTypes.func,
-        passwordConfig: PropTypes.object,
-    };
+interface Error {
+    message: string;
+}
+
+interface PasswordConfig {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+    requireUppercase: boolean;
+}
 
-    static defaultProps = {
+type State = {
+    serverErrorNewPass: JSX.Element|null;
+    serverErrorCurrentPass: JSX.Element|string|null;
+}
+
+type Props = {
+    user?: UserProfile;
+    currentUserId: string;
+    show: boolean;
+    onModalSubmit: (user?: UserProfile) => void;
+    onModalDismissed: () => void;
+    passwordConfig: PasswordConfig;
+    actions: Actions;
```

Comment:
Probably makes sense to embed directly here the `actions` definition as a struct, I don't see the point of split that if we are not going to use that in any other place.

Reply:
Done